packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -232,6 +237,31 @@ Learn more about compiler configuration at https://hardhat.org/config"
   }
 }
 
+function checkUnsupportedRemappings(userConfig: any) {
+  if (typeof userConfig.solidity === "object") {
+    let remappings: boolean;
+    if ("compilers" in userConfig.solidity) {
+      remappings =
+        userConfig.solidity.compilers.filter(
+          ({ settings }: SolcUserConfig) => settings?.remappings !== undefined
+        ).length > 0;
+    } else {
```

Comment:
I think there's another format here, which is `solidity: "x.x.x"`

Reply:
there is, but it's covered by the initial `if (typeof userConfig.solidity === 'object') {...}` check

Reply:
Oops, i didn't noticed that. you are right